proxy/proxy.go

```diff
@@ -0,0 +1,506 @@
+		}
+		var userSig = new(Signal)
+		err = json.Unmarshal(body, userSig)
+		if err != nil {
+			panic(err)
+			log.Println(err)
+		}
+		sigResp, err := node.SendSignal(userSig)
+		js, err := json.Marshal(sigResp)
+		if err != nil {
+			panic(err)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(js)
+	}
+}
+
+func (node *Node) SendRequest(userReq *Request) (reqResp RequestResponse, err error) {
+	req := node.engine.NewRequest(userReq.Topic)
+	resp, err := req.Execute(G.NewMessage().SetData(userReq.Message))
+	if err != nil {
+		log.Println("Echoclient: error", err.Error())
+	}
+
+	var output string
+	if err := resp.Next().GetData(&output); err != nil {
+		log.Println("Echoclient: error", err.Error())
+	} else {
+		log.Println("Echoclient: received", output)
+	}
+	reqResp = RequestResponse{
+		Messages: map[string]interface{}{
+			"result": output,
+		},
+		Length: len(output),
+	}
+	return
+}
+
+func requestHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "POST" {
+		nodeId := req.URL.Path[len("/request/"):]
+		node, _ := nMap.Get(NodeID(nodeId))
+		body, err := ioutil.ReadAll(req.Body)
+		if err != nil {
+			log.Println(err)
+			panic(err)
+		}
+		var userReq = new(Request)
+		err = json.Unmarshal(body, userReq)
+		if err != nil {
+			panic(err)
+			log.Println(err)
+		}
+		reqResp, err := node.SendRequest(userReq)
+		js, err := json.Marshal(reqResp)
+		if err != nil {
+			panic(err)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(js)
+	}
+}
+
+func isAliveHandler(w http.ResponseWriter, req *http.Request) {
+	fmt.Fprintf(w, "I am %s!", "alive")
+}
+
+func CreatePublishSocket(NodeID string) (l net.Listener, err error) {
+	l, err = net.Listen("unix", "/tmp/publish_socket"+NodeID[0:5]+".sock")
+	if err != nil {
+		log.Println("listen error:", err)
+		return
+	}
+	go func() {
+		for {
+			http.HandleFunc("/request/"+NodeID, requestHandler)
+			http.HandleFunc("/signal/"+NodeID, signalHandler)
+			http.HandleFunc("/health_check/"+NodeID, isAliveHandler)
+			err = http.Serve(l, nil)
+		}
+	}()
+	closeOnInterrupt(l)
+	return
+}
+
+func uniqueNodeId(strlen int) (id string) {
+	rand.Seed(time.Now().UTC().UnixNano())
+	const chars = "abcdefghijklmnopqrstuvwxyz0123456789"
+	result := make([]byte, strlen)
+	for i := 0; i < strlen; i++ {
+		result[i] = chars[rand.Intn(len(chars))]
+	}
+	return string(result)
+}
+
+func makeGilmour(connect string) (engine *G.Gilmour, err error) {
+	redis := backends.MakeRedis(connect, "")
+	engine = G.Get(redis)
+	return
+}
+
+func (node *Node) Start() (err error) {
+	if node.engine == nil {
+		return errors.New("Please setup backend engine")
+	}
+	node.engine.Start()
+	if err = node.AddServices(node.services); err != nil {
+		return
+	}
+	if err = node.AddSlots(node.slots); err != nil {
+		return
+	}
+	return
+}
+
+func CreateNode(nodeReq *NodeReq) (node *Node, err error) {
+	node = new(Node)
+	engine, _ := makeGilmour("127.0.0.1:6379")
```

Comment:
The hardcoding will change once we have a config file or some mode by which user can interact
I will pass Gilmour object in params for `CreateNode`